src/library.cpp

```diff
@@ -2587,6 +2605,94 @@ void lammps_scatter_atoms_subset(void *handle, char *name, int type, int count,
   END_CAPTURE
 }
 
+/* ----------------------------------------------------------------------
+   Contributing author: Robert Meissner (TUHH & HZG, Hamburg, germany)
+   gather bonded info for all atoms
+     return it in user-allocated data
+   data will be ordered by atom ID
+     requirement for consecutive atom IDs (1 to N)
+   return atom-based values in 1d data, ordered by atom ID
+     e.g. b[0],b[1],x[0][2],x[1][0],x[1][1],x[1][2],x[2][0],...
+     data must be pre-allocated by caller to correct length
+     correct length = Nbonds, as queried by get_nbonds()
+   method:
+     alloc and zero Nbond length vector
+     loop over Nlocal to fill vector with my values
+     Allreduce to sum vector into data across all procs
+------------------------------------------------------------------------- */
+
+void lammps_gather_bonds(void *handle, void *data)
+{
+  LAMMPS *lmp = (LAMMPS *) handle;
+
+  BEGIN_CAPTURE
+  {
+#if defined(LAMMPS_BIGBIG)
+    lmp->error->all(FLERR,"Library function lammps_gather_atoms() "
+                    "is not compatible with -DLAMMPS_BIGBIG");
+#else
+    int i,j;
+
+    // error if tags are not defined or not consecutive
+    // NOTE: test that name = image or ids is not a 64-bit int in code?
+
+    int flag = 0;
+    if (lmp->atom->tag_enable == 0 || lmp->atom->tag_consecutive() == 0)
+      flag = 1;
+    if (lmp->atom->natoms > MAXSMALLINT) flag = 1;
+    if (flag) {
+      if (lmp->comm->me == 0)
+        lmp->error->warning(FLERR,"Library error in lammps_gather_bonds");
+      return;
+    }
+    
+    int natoms = static_cast<int> (lmp->atom->natoms);
+    int nbonds = static_cast<int> (lmp->atom->nbonds);
```

Comment:
You need to check for overflows here. You may have more bonds than atoms and the number of bonds may double when `newton_bond` is off.

Reply:
Didn't I took care of this by using `if (!lmp->force->newton_bond) nbonds *= 2;` a line after I get `int nbonds = ...`?

Reply:
the if statement only accounts for the number of stored bond information items, it does not protect it from overflowing a 32-bit int. In fact the doubling makes it even more likely. Overall, it is probably better to stick to bigint data types inside of library.cpp where possible and only use or convert to int what is accessible to the calling program.

Reply:
I have now checked if 'nbonds' will overflow when newton is switched off. But I was not sure if I should check if it generally overflows when newton is turned on. Isn't it already checked when LAMMPS initializes the system?